scripts/ci-entrypoint.sh

```diff
@@ -279,22 +285,14 @@ install_addons() {
 
 copy_secret() {
     # point at the management cluster
-    unset KUBECONFIG
-    "${KUBECTL}" get secret "${CLUSTER_NAME}-control-plane-azure-json" -o jsonpath='{.data.control-plane-azure\.json}' | base64 --decode >azure_json
+    "${KUBECTL}" --kubeconfig "${REPO_ROOT}/${KIND_CLUSTER_NAME}.kubeconfig" get secret "${CLUSTER_NAME}-control-plane-azure-json" -o jsonpath='{.data.control-plane-azure\.json}' | base64 --decode >azure_json
 
-    # set KUBECONFIG back to the workload cluster
-    export KUBECONFIG="${KUBECONFIG:-${PWD}/kubeconfig}"
```

Comment:
shouldn't this be replaced by adding `--kubeconfig ${KUBECONFIG:-${PWD}/kubeconfig}" on all the kubectl commands below that expect to be pointing at the workload cluster?

Reply:
So what's happening is that in the current implementation we're assuming that the default kubeconfig context is pointing at the kind mgmt cluster. Then we explicitly set `KUBECONFIG` to point to the workload cluster in `install_addons`. Later in that flow we want to point back to the mgmt cluster to grab the workload cluster's kubeconfig secret. Now that we have an explicit reference to the mgmt cluster (via `${REPO_ROOT}/${KIND_CLUSTER_NAME}.kubeconfig`) we can do that explicitly (see the changed L288 above), and we don't have to `unset` then `export` again.